server/services/permissionsService.ts

```diff
@@ -0,0 +1,166 @@
+import { HmppsStatusCode } from '../data/enums/hmppsStatusCode'
+import { isActiveCaseLoad, isInUsersCaseLoad, userCanEdit, userHasAllRoles, userHasRoles } from '../utils/utils'
+import { Role } from '../data/enums/role'
+import Prisoner from '../data/interfaces/prisonerSearchApi/Prisoner'
+import { HmppsUser } from '../interfaces/HmppsUser'
+import UserService from './userService'
+import getOverviewAccessStatusCode from './utils/permissions/getOverviewAccessStatusCode'
+
+interface PermissionItem {
+  view: boolean
+  edit?: boolean
+}
+
+export interface Permissions {
+  accessCode: HmppsStatusCode
+  courtCases?: PermissionItem
+  money?: PermissionItem
+  adjudications?: PermissionItem
+  visits?: PermissionItem
+  incentives?: PermissionItem
+  category?: PermissionItem
+  calculateReleaseDates?: PermissionItem
+  caseNotes?: PermissionItem
+  keyWorker?: PermissionItem
+  appointment?: PermissionItem
+  useOfForce?: PermissionItem
+  activity?: PermissionItem
+  pathfinder?: PermissionItem
+  soc?: PermissionItem
+  offenderCategorisation?: PermissionItem
+  probationDocuments?: PermissionItem
+}
+
+export default class PermissionsService {
+  constructor(private readonly userService: UserService) {}
+
+  public async getOverviewPermissions(user: HmppsUser, prisoner: Prisoner, clientToken?: string): Promise<Permissions> {
+    const { userRoles } = user
+
+    const accessCode = getOverviewAccessStatusCode(user, prisoner)
+    if (accessCode !== HmppsStatusCode.OK) return { accessCode }
+
+    const isPrisonerInCaseLoad = isInUsersCaseLoad(prisoner.prisonId, user)
+
+    const staffRoles = await this.userService.getStaffRoles(clientToken, user)
+
+    return {
+      accessCode,
+      courtCases: {
+        view: userHasRoles([Role.ReleaseDatesCalculator], userRoles),
+        edit: userHasRoles([Role.AdjustmentsMaintainer], userRoles),
+      },
+      money: {
+        view: isPrisonerInCaseLoad,
+      },
+      adjudications: {
+        view: isPrisonerInCaseLoad || userHasRoles([Role.PomUser, Role.ReceptionUser], userRoles),
+      },
+      visits: {
+        view: isPrisonerInCaseLoad || userHasRoles([Role.PomUser, Role.ReceptionUser], userRoles),
+      },
+      incentives: {
+        view: isPrisonerInCaseLoad || userHasRoles([Role.GlobalSearch], userRoles),
+      },
+      category: {
+        view: false,
+        edit: userHasRoles(
+          [
+            Role.CreateRecategorisation,
+            Role.ApproveCategorisation,
+            Role.CreateRecategorisation,
+            Role.CategorisationSecurity,
+          ],
+          userRoles,
+        ),
+      },
+      calculateReleaseDates: {
+        view: false,
+        edit: userHasRoles([Role.ReleaseDatesCalculator], user.userRoles),
+      },
+      caseNotes: {
+        view: false,
```

Comment:
(same with the other hardcoded `view: false` instances)

Reply:
I had assumed that they would all require a view at some point further into this process and was trying to avoid a type where everything is optional as it is pointless. Maybe they won't though so can take out for now